File: ZeekoBlog/wwwroot/ts/Article.ts
export class Article implements IArticle {
    id?: number;
    title: string = '';
    summary: string = '';
    content: string = '';
    lastEdited: Date = new Date();

    constructor(obj?: IArticle) {
        Object.assign(this, obj);
    }
}

interface IArticle {
    id?: number;
    title?: string;
    summary?: string;
    content?: string;
    lastEdited?: Date;
}

export enum ArticleDocType {
    Markdown, AsciiDoc, Raw
}

export interface IArticlePostDto {
    title: string;
    summary: string;
    content: string;
    docType: ArticleDocType;
}

File: ZeekoBlog.Core/Models/Article.cs
using System;
using System.Collections.Generic;

namespace ZeekoBlog.Core.Models
{
    public class Article
    {
        public int Id { get; set; }
        public BlogUser BlogUser { get; set; }
        public string Title { get; set; }
        public string Summary { get; set; }
        public string Content { get; set; }
        public DateTime LastEdited { get; set; }
        /// <summary>
        /// 文章中涉及到的编程语言
        /// </summary>
        public string Languages { get; set; } = string.Empty;
        /// <summary>
        /// 文章类型，为了兼容以前的文章，默认为 Markdown
        /// </summary>
        public ArticleDocType DocType { get; set; } = ArticleDocType.Markdown;

        public string RenderedContent { get; set; } = string.Empty;
        public string RenderedSummary { get; set; } = string.Empty;
        public List<TOCItem> TOCList { get; set; } = new List<TOCItem>();
        public DateTime Created { get; set; }
        public bool SoftDelete { get; set; } = false;
    }

    public enum ArticleDocType
    {
        Markdown, AsciiDoc, Raw
    }
}
